File: Leetcode/python/sol20.py
# https://leetcode.com/problems/valid-parentheses/description/

class Solution:
    def isValid(self, s: str) -> bool:
        if len(s) % 2 != 0:
            return False
        stack = []
        brackets = {']':'[', '}':'{', ')':'('}
        for i in s:
            if i in brackets:
                if stack and stack[-1] == brackets[i]:
                    stack.pop()
                else:
                    return False
            else:
                stack.append(i)

        return len(stack) == 0

File: Leetcode/javascript/sol20.js


//https://leetcode.com/problems/valid-parentheses/description/
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
    var stack = [];
    const brackets = {
        '}':'{',
        ')':'(',
        ']':'['
    };

    for(const char of s){
        if (char in brackets){
            if (stack.length != 0){
                var last_element = stack[stack.length -1];
                if (last_element == brackets[char]){
                    stack.pop();
                }else{
                    return false;
               a }
            }else{
                return false;
            }

        }else{
            stack.push(char);
        }
    }
    return stack.length == 0;
};